src/coreclr/vm/codeman.cpp

```diff
@@ -1540,7 +1525,34 @@ void EEJitManager::SetCpuInfo()
     CPUCompileFlags.Set64BitInstructionSetVariants();
     CPUCompileFlags.EnsureValidInstructionSetSupport();
 
-#if defined(TARGET_X86) || defined(TARGET_AMD64)
+    _ASSERTE(!CPUCompileFlags.IsSet(InstructionSet_VectorT128));
+
+#if defined(TARGET_ARM64)
+    if (CPUCompileFlags.IsSet(InstructionSet_AdvSimd))
+    {
+        CPUCompileFlags.Set(InstructionSet_VectorT128);
+    }
+#elif defined(TARGET_X86) || defined(TARGET_AMD64)
+
+    _ASSERTE(!CPUCompileFlags.IsSet(InstructionSet_VectorT512));
+    _ASSERTE(!CPUCompileFlags.IsSet(InstructionSet_VectorT256));
```

Comment:
nit: Put `VectorT256` first

Can we also add a comment explaining why this logic needs to be here below `EnsureValidInstructionSetSupport` and the `CLRConfig::Enable*` checks